File: src/gestorAplicacion/enrutadorHFC/Dispositivo.java
package gestorAplicacion.enrutadorHFC;

import gestorAplicacion.host.Cliente;

import java.io.Serializable;
import java.util.ArrayList;

public class Dispositivo implements Serializable{

	//CLASE IMPORTANTE PARA LA FUNCIONALIDAD DE VISUALIZAR DISPOSITIVOS, SIN EMBARGO TAMBIÉN TIENE IMPLICACIONES EN OTRAS EN PROCESOS SECUNDARIOS (TEST--REPORTE--MEJORA TU PLAN)

	//ATRIBUTOS
	private Router modem;
	private String ipAsociada;
	private String nombre;
  	private String generacion;
	private static ArrayList<Dispositivo> dispositivosTotales = new ArrayList<>();
  

	//CONSTRUCTORS
	public Dispositivo(Router modem,String nombre,String generacion) {
	    this.modem = modem;
	    this.nombre = nombre;
	    ipAsociada=this.modem.getIP();
      	this.generacion=generacion;
      	dispositivosTotales.add(this);
	}

    //MÉTODOS

    //METODO DE ESTATICO---FUNCIONALIDAD TEST
	public static String desconectarDispositivo(Cliente cliente,Dispositivo dispositivo) {
	    if(cliente.getModem().verificarDispositivos(dispositivosTotales,cliente.getModem()).contains(dispositivo)){
	      dispositivo.setIpAsociada("");
	      return "Dispositivo desconectado correctamente.";
	    }else{
	      return "Dispositivo inválido, intente de nuevo.";
	    } 
	}

	//METODO TOSTRING
	public String toString() {
		return "Dispositivo: "+ this.nombre + "\nDirección IP: "+ this.ipAsociada+"\nGeneración: "+this.generacion;
	}


	//GETTERS Y SETTERS
	public Router getModem() {
		return modem;
	}

	public void setModem(Router modem) {
		this.modem = modem;
	}

	public String getIpAsociada() {
		return ipAsociada;
	}

	public void setIpAsociada(String ipAsociada) {
		this.ipAsociada = ipAsociada;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public String getGeneracion() {
		return generacion;
	}

	public void setGeneracion(String generacion) {
		this.generacion = generacion;
	}

	public static ArrayList<Dispositivo> getDispositivosTotales() {
		return dispositivosTotales;
	}

	public static void setDispositivosTotales(ArrayList<Dispositivo> dispositivosTotales) {
		Dispositivo.dispositivosTotales = dispositivosTotales;
	}
}

File: Python/src/gestorAplicacion/enrutadorHFC/dispositivo.py
class Dispositivo:

  # //CLASE IMPORTANTE PARA LA FUNCIONALIDAD DE VISUALIZAR DISPOSITIVOS, SIN EMBARGO TAMBIÉN TIENE IMPLICACIONES EN OTRAS EN PROCESOS SECUNDARIOS (TEST--REPORTE--MEJORA TU PLAN)
  
  # ATRIBUTO ESTÁTICO
  _dispositivosTotales=[]

  # CONSTRUCTORES
  def __init__(self, modem, nombre, generacion):

    # //ATRIBUTOS
    self._modem=modem
    self._nombre=nombre
    self._ipAsociada=modem.getIP()
    self._generacion=generacion
    Dispositivo._dispositivosTotales.append(self)
  
  # //MÉTODOS

  # //METODO DE ESTATICO---FUNCIONALIDAD TEST

  @classmethod
  def desconectarDispositivo(cls, cliente, dispositivo):
    if (cliente.getModem().verificarDispositivos(Dispositivo._dispositivosTotales,cliente.getModem()).contains(dispositivo) ):
      dispositivo.setIpAsociada("")
      return "Dispositivo desconectado correctamente."
    else:
      return "Dispositivo inválido, intente de nuevo."
  
  # //METODO TOSTRING
  def __str__(self):
    return f"Dispositivo {self._nombre} \nDirección IP: {self._ipAsociada} \nGeneración: {self._generacion}"
    
  # //GETTERS Y SETTERS
  def getModem(self):
    return self._modem
  
  def setModem(self, modem):
    self._modem=modem
    
  def getIpAsociada(self):
    return self._ipAsociada
  
  def setIpAsociado(self, ipAsociada):
    self._ipAsociada=ipAsociada

  def getNombre(self):
    return self._nombre
  
  def setNombre(self, nombre):
    self._nombre=nombre

  def getGeneracion(self):
    return self._generacion
  
  def setGeneracion(self, generacion):
    self._generacion=generacion

  @classmethod
  def getDispositivosTotales(cls):
    return cls._dispositivosTotales
  
  @classmethod
  def setDispositivosTotales(cls, _dispositivosTotales):
    cls._dispositivosTotales=_dispositivosTotales
  